nomad/blocked_evals.go

```diff
@@ -133,6 +146,14 @@ func (b *BlockedEvals) Block(eval *structs.Evaluation) {
 		return
 	}
 
+	// Check if the eval missed an unblock while it was in the scheduler at an
+	// older index.
```

Comment:
I would add some more comment as to how/why this is possible. "The scheduler could have been invoked with a snapshot of state that was prior to additional capacity being added or allocations becoming terminal."